node/resolvers/mutations/index.ts

```diff
@@ -71,88 +76,122 @@ export const Mutation = {
     }
 
-    const email = sessionData.namespaces.profile.email.value
-    const {
-      role: { slug },
-    } = storefrontPermissions
+    try {
+      let quoteBySeller: SellerQuoteMap = {}
 
-    const {
-      organization: { value: organizationId },
-      costcenter: { value: costCenterId },
-    } = sessionData.namespaces['storefront-permissions']
+      if (settings?.adminSetup.quotesManagedBy === 'SELLER') {
+        const sellerItems = items.filter(
+          ({ seller }) => seller && seller !== '1'
+        )
 
-    const now = new Date()
-    const nowISO = now.toISOString()
-    const expirationDate = new Date()
+        quoteBySeller = await splitItemsBySeller({
+          ctx,
+          items: sellerItems,
+        })
+      }
 
-    expirationDate.setDate(
-      expirationDate.getDate() + (settings?.adminSetup?.cartLifeSpan ?? 30)
-    )
-    const expirationDateISO = expirationDate.toISOString()
+      const hasSellerQuotes = Object.keys(quoteBySeller).length
 
-    const status = sendToSalesRep ? 'pending' : 'ready'
-    const lastUpdate = nowISO
-    const updateHistory = [
-      {
-        date: nowISO,
-        email,
+      const parentQuoteItems = hasSellerQuotes
+        ? items.filter(
+            (item) =>
+              !Object.values(quoteBySeller).some((quote) =>
+                quote.items.some(createItemComparator(item))
+              )
+          )
+        : items
+
+      const quoteCommonFields = {
+        sessionData,
+        storefrontPermissions,
+        segmentData,
+        settings,
+        referenceName,
         note,
-        role: slug,
-        status,
-      },
-    ]
-
-    const salesChannel: string = segmentData?.channel
-
-    const quote = {
-      costCenter: costCenterId,
-      creationDate: nowISO,
-      creatorEmail: email,
-      creatorRole: slug,
-      expirationDate: expirationDateISO,
-      items,
-      lastUpdate,
-      organization: organizationId,
-      referenceName,
-      status,
-      subtotal,
-      updateHistory,
-      viewedByCustomer: !!sendToSalesRep,
-      viewedBySales: !sendToSalesRep,
-      salesChannel,
-    }
+        sendToSalesRep,
+      }
 
-    try {
-      const data = await masterdata
-        .createDocument({
-          dataEntity: QUOTE_DATA_ENTITY,
-          fields: quote,
-          schema: SCHEMA_VERSION,
-        })
-        .then((res: any) => res)
+      // We believe that parent quote should contain the overall subtotal.
+      // If for some reason it is necessary to subtract the subtotal from
+      // sellers quotes, we can use the adjustedSubtotal below, assigning
+      // it to subtotal in createQuoteObject -> `subtotal: adjustedSubtotal`
+      //
+      // const adjustedSubtotal = hasSellerQuotes
+      //   ? Object.values(quoteBySeller).reduce(
+      //       (acc, quote) => acc - quote.subtotal,
+      //       subtotal
+      //     )
+      //   : subtotal
+      const parentQuote = createQuoteObject({
```

Comment:
If there are no items from the marketplace, the parent quote should not be created.

Reply:
There needs to be a link between the sellers quotes and the "parent" quote, as we said in [these comment](https://github.com/vtex-apps/b2b-quotes-graphql/pull/59#discussion_r1889227674).